data_steward/utils/participant_summary_requests.py

```diff
@@ -62,40 +71,168 @@ def get_access_token():
     return access_token
 
 
-def get_participant_data(url, headers):
+def get_participant_data(api_project_id: str,
+                         params: Dict,
+                         expected_fields: List[str] = None) -> List[Dict]:
     """
     Fetches participant data via ParticipantSummary API
 
-    :param url: the /ParticipantSummary endpoint to fetch information about the participant
-    :param headers: the metadata associated with the API request and response
+    :param api_project_id: RDR project id when PS API rests
+    :param params: the fields and their values
+    :param expected_fields: filter participants not containing any of the fields.
+        Use only if API cannot filter fields via params. If unspecified, fetches all participants
 
     :return: list of data fetched from the ParticipantSummary API
     """
+    # Base /ParticipantSummary endpoint to fetch information about the participant
+    url = f'https://{api_project_id}.appspot.com/rdr/v1/ParticipantSummary'
 
     done = False
     participant_data = []
-    original_url = url
-    next_url = url
+
+    token = get_access_token()
+
+    headers = {
+        'content-type': 'application/json',
+        'Authorization': f'Bearer {token}'
+    }
+
+    session = Session()
 
     while not done:
-        resp = requests.get(next_url, headers=headers)
+        resp = session.get(url, headers=headers, params=params)
         if not resp or resp.status_code != 200:
-            print(f'Error: API request failed because {resp}')
+            LOGGER.warning(f'Error: API request failed because {resp}')
         else:
+            LOGGER.info(f'Fetching data from PS API using params:{params}')
             r_json = resp.json()
-            participant_data += r_json.get('entry', {})
+            participant_data += r_json.get(
+                'entry', {}
+            ) if not expected_fields else [
+                # filter out participants who do not have all fields in expected_fields
+                row
+                for row in r_json.get('entry', {})
+                if row.get('resource', {}).keys() &
+                set(expected_fields) == set(expected_fields)
```

Comment:
The full expression is `if row.get('resource', {}).keys() & set(expected_fields) == set(expected_fields)`, it just looks misleading due to yapf formatting. This evaluates to `True` only when the fetched row contains the `expected_fields`. This is helpful because a lot of participants don't have the `digitalHealthSharingStatus` field.